include/irr/asset/filters/CSummedAreaTableImageFilter.h

```diff
@@ -93,10 +115,13 @@ class CSummedAreaTableImageFilter : public CMatchedSizeInOutImageFilterCommon, p
 			if (state->scratchMemoryByteSize < state_type::getRequiredScratchByteSize(state->inImage))
 				return false;
 
+			if (state->inMipLevel != state->outMipLevel)
+				return false;
+
 			if (asset::getFormatChannelCount(outFormat) != asset::getFormatChannelCount(inFormat))
 				return false;
 
-			if (asset::getFormatClass(state->inImage->getCreationParameters().format) < asset::getFormatClass(outFormat))
+			if (asset::getFormatClass(inFormat) >= asset::getFormatClass(outFormat))
```

Comment:
`>=` comparison is not safe... you'd actually need to use one of the functions that give the per-channel bit-depth and compare that the channel bit-depth is larger.

Reply:
we don't have such a function probably, I only see something what returns max value for the channel

Reply:
TODO!

Reply:
@devshgraphicsprogramming @Crisspl we probably need to write such one